### PR TITLE
feat: support other text formats

### PR DESCRIPTION
**Pull request by hrik2001**
This PR is to add file extension validation for text URL.
_(This PR is related to the issue #1166.)_

**Note:**
The protobuf test for text URL didn't test for remote URL (it only had local URL), moreover, it didn't check if the generated protobuf has `text_url` in it or not. I added both these features and also marked the test `slow` and `internet`, just how it is being done in other tests (for example test for `audio_url`). Do let me know if such change is not desired (as this change was not part of the issue), then I will add a commit removing that and squash.

### CODE REVIEW

**Comment by hrik2001** [comment]
Looks like my changes broke other tests, hence converting this PR to a draft for the time being, until it's all fixed! :D

**Review by JohannesMessner — APPROVED** [review]
Looks good to me, once the tests pass ;)

**Review by JohannesMessner — DISMISSED** [review]
Approved accidentally, the test need to pass of course, but looks good so far!

File: docarray/typing/url/text_url.py [hunk 1]
```diff
@@ -11,6 +18,27 @@ class TextUrl(AnyUrl):
...
+    ):
+        import os
+        from urllib.parse import urlparse
+
+        url = super().validate(value, field, config)  # basic url validation
+        path = urlparse(url).path
+        ext = os.path.splitext(path)[1][1:].lower()
+
+        # pass test if extension is valid or no extension
+        has_text_extension = ext in TEXT_FILE_FORMATS or ext == ''
```
Comment (samsja) [inline docarray/typing/url/text_url.py:36]:
```suggestion
        has_valide_text_extension = ext in TEXT_FILE_FORMATS or ext == ''
```

File: tests/units/typing/url/test_text_url.py [hunk 2]
```diff
@@ -57,5 +77,17 @@ def test_json_schema():
 
 
 def test_dump_json():
```
Comment (samsja) [inline tests/units/typing/url/test_text_url.py:79]:
```suggestion
@pytest.mark.internet
def test_dump_json():
```

**Review by samsja — APPROVED** [review]

**Pull request merged**